openmc/mgxs_library.py

```diff
@@ -379,200 +436,587 @@ def tabular_legendre(self, tabular_legendre):
     @num_polar.setter
     def num_polar(self, num_polar):
         # Make sure we have positive ints
-        check_value("num_polar", num_polar, Integral)
-        check_greater_than("num_polar", num_polar, 0)
+        check_value('num_polar', num_polar, Integral)
+        check_greater_than('num_polar', num_polar, 0)
         self._num_polar = num_polar
 
     @num_azimuthal.setter
     def num_azimuthal(self, num_azimuthal):
-        check_value("num_azimuthal", num_azimuthal, Integral)
-        check_greater_than("num_azimuthal", num_azimuthal, 0)
+        check_value('num_azimuthal', num_azimuthal, Integral)
+        check_greater_than('num_azimuthal', num_azimuthal, 0)
         self._num_azimuthal = num_azimuthal
 
     @total.setter
     def total(self, total):
-        if self._representation is 'isotropic':
-            shape = (self._energy_groups.num_groups,)
-        elif self._representation is 'angle':
-            shape = (self._num_polar, self._num_azimuthal,
-                     self._energy_groups.num_groups)
+        """This method sets the total cross section by performing a
+        deep-copy of the provided ndarray. If the angular
```

Comment:
And for all of the other properties below. I think these are actually our first property docstrings in the API. Since the docstrings for properties don't show up in Jupyter Notebook as nicely as they do for normal methods, I would recommend putting all of the info for each property in the "Attributes" section of the `XSData` class docstring (and perhaps even removing it from the properties themselves).